frontend/src/components/common/ConfirmDialog.tsx

```diff
@@ -30,6 +35,44 @@ export function ConfirmDialog(props: ConfirmDialogProps) {
     }
   }, []);
 
+  if (props.checkboxDescription) {
+    return (
+      <div>
+        <MuiDialog
+          open={open}
+          onClose={handleClose}
+          aria-labelledby="alert-dialog-title"
+          aria-describedby="alert-dialog-description"
+        >
+          <DialogTitle id="alert-dialog-title">{title}</DialogTitle>
+          <DialogContent ref={focusedRef}>
+            <DialogContentText id="alert-dialog-description">{description}</DialogContentText>
+            <Box
+              sx={{
+                display: 'flex',
+                alignItems: 'center',
+                marginTop: '10px',
+              }}
+            >
+              <DialogContentText id="alert-dialog-description">
+                {props.checkboxDescription}
+              </DialogContentText>
+              <Checkbox onChange={() => setCheckboxClicked(!checkboxClicked)} />
+            </Box>
+          </DialogContent>
+          <DialogActions>
+            <Button onClick={handleClose} color="primary">
+              {t('Cancel')}
+            </Button>
+            <Button disabled={!checkboxClicked} onClick={onConfirmationClicked} color="primary">
+              {t('I Agree')}
```

Comment:
I know I asked for "I agree", but I didn't remember this was in the context of an existing ConfirmDialog.
The ConfirmDialog should work similar whether you have a checkbox or not. I think it's better to add the checkbox (when existing) to the original code, rather than having two different ways of rendering the dialog. And the buttons should remain with the original labels, unless we want to provide properties for those too.

Reply:
Done! made some changes to the logic for the dialog which should allow us to just provide a description and then the smaller section of that description and the checkbox should render